packages/library-base/src/commonMain/kotlin/io/realm/kotlin/Configuration.kt

```diff
@@ -165,6 +165,20 @@ public interface Configuration {
      */
     public val initialDataCallback: InitialDataCallback?
 
+    /**
+     * Enums describing describing where the Realm exists. [FULL] implies that the realm resides on
+     * disk while [MEM_ONLY] implies that the realm will reside in memory. Note that the content
+     * of the realm is not persistent if it is closed while having a durability of [MEM_ONLY].
+     */
+    public enum class Durability {
+        FULL, MEM_ONLY
+    }
+
+    /**
+     * Describes whether the realm should reside in memory or on disk.
+     */
+    public val durability: Durability
```

Comment:
We should also add tests to `RealmConfigurationTests` and `SyncConfigurationTests` that verify the value of this.

It also looks like you didn't add support for SyncConfigurations (judging from CI), so having that test there would also catch that.